component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -763,6 +788,15 @@ private void redirectToEmailAddressReqPage(HttpServletResponse response, Authent
             String emailAddressReqPage = getEmailAddressRequestPage(context, emailOTPParameters);
             try {
                 String url = getRedirectURL(emailAddressReqPage, queryParams);
+                if (isEmailUpdateFailed(context)) {
+                    url = url + EmailOTPAuthenticatorConstants.RETRY_PARAMS;
+                    if (context.getProperty(EmailOTPAuthenticatorConstants.PROFILE_UPDATE_FAILURE_REASON) != null) {
+                        String failureReason = String.valueOf(
+                                context.getProperty(EmailOTPAuthenticatorConstants.PROFILE_UPDATE_FAILURE_REASON));
+                        String urlEncodedFailureReason = URLEncoder.encode(failureReason, CHAR_SET_UTF_8);
+                        url = url + EmailOTPAuthenticatorConstants.ERROR_MESSAGE_DETAILS +  urlEncodedFailureReason;
```

Comment:
use FrameworkUtils.appendQueryParamToURL

Reply:
Fixed this issue